dashboard/app/views/layouts/application.html.haml

```diff
@@ -66,8 +66,13 @@
     - else
       - unless Rails.env.production?
         = render partial: 'layouts/environment_tag'
-    - if current_user&.student? && (current_user.age.nil? || (current_user.us_state.nil? && Policies::Lti.lti?(current_user)))
-      = render partial: 'layouts/student_information_interstitial'
+    - is_student = current_user&.student?
+    - is_in_us = request.country.nil? || ['US', 'RD'].include?(request.country.to_s.upcase)
+    - under_age_13 = current_user.age.nil? || current_user.age.to_i < 13 if is_student
+    - retrieve_info_for_lti = current_user&.us_state.nil? && Policies::Lti.lti?(current_user) if is_student
+    - retrieve_info_for_cap = current_user&.us_state.nil? && is_in_us && under_age_13 && Policies::ChildAccount.personal_account?(current_user) if is_student
+    - if is_student && (current_user.age.nil? || retrieve_info_for_lti || retrieve_info_for_cap)
+      = render partial: 'layouts/student_information_interstitial', locals: {retrieve_info_for_lti: retrieve_info_for_lti, retrieve_info_for_cap: retrieve_info_for_cap}
```

Comment:
There is a lot of logic here and it can get hard to read. Maybe, to improve readability you can use an if statement. Let me know if it feels more readable.
```
- if current_user&.student?
    - under_age_13 = current_user.age.nil? || current_user.age.to_i < 13
    - retrieve_info_for_lti = current_user&.us_state.nil? && Policies::Lti.lti?(current_user)
    - retrieve_info_for_cap = current_user&.us_state.nil? && is_in_us && under_age_13 && Policies::ChildAccount.personal_account?(current_user)
    - missing_student_data? = (current_user.age.nil? || retrieve_info_for_lti || retrieve_info_for_cap)
  
 - if missing_student_data? 
     = render partial: 'layouts/student_information_interstitial', locals: {retrieve_info_for_lti: retrieve_info_for_lti, retrieve_info_for_cap: retrieve_info_for_cap}
```

Reply:
We can use helper methods for these: https://www.rubyguides.com/2020/01/rails-helpers
Please check: https://github.com/code-dot-org/code-dot-org/pull/58003/commits/cd5635ef63b7c176d4904ca03dfc27c3569620f8